cloud_functions/ads_policy_monitor/google_ads.py

```diff
@@ -95,14 +108,65 @@ def run_gaarf_report(payload: models.Payload,
                                  report_fetcher=report_fetcher,
                                  customer_ids=payload.customer_ids)
 
-    path = f'gaql/{report_config.gaql_filename}'
+    if report_config.is_asset_report:
+        all_reports = []
+        for gaql_filename in report_config.gaql_filenames:
+            path = f'gaql/{gaql_filename}'
+            all_reports.append(
+                run_query_from_file(
+                    query_path=path,
+                    report_fetcher=report_fetcher,
+                    customer_ids=payload.customer_ids,
+                ))
+        return combine_assets_reports(all_reports)
+
+    path = f'gaql/{report_config.gaql_filenames}'
     return run_query_from_file(
         query_path=path,
         report_fetcher=report_fetcher,
         customer_ids=payload.customer_ids,
     )
 
 
+def combine_assets_reports(gaarf_reports: List[GaarfReport]) -> GaarfReport:
+    """Combine assets reports.
+
+    Args:
+        gaarf_reports: a list of gaarf reports.
+
+    Returns:
+        A GAARF report.
+    """
+    logger.info('Combining Assets reports')
+    combined_assets_report = pd.DataFrame()
```

Comment:
Is this needed? You override this later on.

Reply:
Yes, otherwise it complains:
UnboundLocalError: local variable 'combined_assets_report' referenced before assignment

I don't override it, I append to it (I use concat while referencing this same dataframe)